src/util/helperUtil.ts

```diff
@@ -80,3 +81,60 @@ export function clearDataSavedInGlobalState(context: vscode.ExtensionContext) {
 	context.globalState.update('selectedProject', undefined);
 }
 
+/**
+ * Traverse upwards to check for target/build directory and its corresponding sibling build file (pom.xml or build.gradle).
+ * @param filePath The file path to start the search from.
+ * @returns true if sibling file found, false otherwise.
+ */
+export async function checkSiblingFilesInTargetOrBuildParent(filePath: string): Promise<boolean> {
+
+	try {
+		let currentDir = path.dirname(filePath);  // Start with the directory of the provided file
+
+		// Traverse upwards through the directories
+		while (currentDir !== path.parse(currentDir).root) {
+			let outputDir: string | undefined;
+			let siblingFilePath: string | undefined;
+
+			// Determine the file type to look for (either pom.xml or build.gradle)
+			if (filePath.endsWith('pom.xml')) {
+				outputDir = 'target';  // For Maven projects, look for target directory
+				siblingFilePath = path.join(currentDir, 'pom.xml');
+			} else if (filePath.endsWith('build.gradle') || filePath.endsWith('settings.gradle')) {
+				outputDir = 'build';  // For Gradle projects, look for build directory
+				siblingFilePath = path.join(currentDir, 'build.gradle');
+			} else {
+				console.log("Invalid file type. Only 'pom.xml' or 'build.gradle' are supported.");
+				return false;
+			}
```

Comment:
The logic for setting up the dir check values can be done outside the while loop. I would recommend something like this:
```suggestion
		let outputDir: string | undefined;
		let siblingFileType: string | undefined;

		// Determine the file type to look for (either pom.xml or build.gradle)
		if (filePath.endsWith('pom.xml')) {
			outputDir = 'target';  // For Maven projects, look for target directory
			siblingFileType = 'pom.xml';
		} else if (filePath.endsWith('build.gradle') || filePath.endsWith('settings.gradle')) {
			outputDir = 'build';  // For Gradle projects, look for build directory
			siblingFileType = 'build.gradle';
		} else {
			console.debug("Invalid file type. Only 'pom.xml' or 'build.gradle' are supported.");
			return false;
		}
		
	       while (currentDir !== path.parse(currentDir).root) {
	           let siblingFilePath = path.join(currentDir, siblingFileType);
			
```

Reply:
addressed